src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableList.cs

```diff
@@ -286,8 +286,9 @@ public static int LastIndexOf<T>(this IImmutableList<T> list, T item, int startI
         {
             Requires.NotNull(list, nameof(list));
 
-            if (list.Count == 0 && startIndex == 0)
+            if (list.Count == 0)
```

Comment:
Both changes are potentially breaking changes. What use cases is this PR meant to address? Could you share an example (or better create an issue so that we can debate potential actions?).

I'm assuming that this has something to do with `ImmutableList<T>` throwing ArgumentOutOfRangeException for certain inputs, but do note that this is changing an extension method acting on an interface.

Reply:
@eiriktsarpalis : Sorry for late reply. I've created a discussion with the PR all together for you to see at my 1st post.

Reply:
Thanks. Let's close this PR for now until a resolution is reached in #58977